contracts/crashpunks-v2-friedger.clar

```diff
@@ -0,0 +1,345 @@
+;; Interface definitions
+(impl-trait .nft-trait.nft-trait)
+(impl-trait .operable.operable)
+;; (impl-trait .transferable.transferable)
+
+;; contract variables
+
+(define-data-var administrator principal 'SP3N4AJFZZYC4BK99H53XP8KDGXFGQ2PRSQP2HGT6)
+
+
+;; TODO: MAKE SURE THIS MINT COUNTER IS CORRECT. SHOULD BE THE MINT-COUNTER FROM V1. DOUBLE CHECK IF OFF BY 1 ERROR
+(define-data-var mint-counter uint u5721)
+
+;; addresses to receive mint fee
+(define-data-var collection-mint-addresses (list 4 principal) (list))
+;; percent mint fee each address receives
+(define-data-var collection-mint-shares (list 4 uint) (list))
+
+;; addresses to receive royalty fees
+;; first element at this list is ignored because the nft owner will be used
+(define-data-var collection-royalty-addresses (list 10 principal) (list))
+;; percent royalty fee each address receives
+(define-data-var collection-royalty-shares (list 10 uint) (list))
+
+;; TODO: update this
+(define-data-var base-uri (string-ascii 80) "ipfs://Qmad43sssgNbG9TpC6NfeiTi9X6f9vPYuzgW2S19BEi49m/{id}")
+(define-data-var metadata-frozen bool false)
+
+;; constants
+(define-constant percentage-with-twodp u10000000000)
+
+;; 50 stx
+(define-constant MINT-PRICE u50000000)
+
+(define-constant token-name "crashpunks-v2")
+(define-constant token-symbol "CPS-v2")
+(define-constant COLLECTION-MAX-SUPPLY u9216)
+
+(define-constant ERR-METADATA-FROZEN (err u101))
+(define-constant ERR-COULDNT-GET-V1-DATA (err u102))
+(define-constant ERR-COULDNT-GET-NFT-OWNER (err u103))
+(define-constant ERR-PRICE-WAS-ZERO (err u104))
+(define-constant ERR-NFT-NOT-LISTED-FOR-SALE (err u105))
+(define-constant ERR-PAYMENT-ADDRESS (err u106))
+(define-constant ERR-NFT-LISTED (err u107))
+(define-constant ERR-COLLECTION-LIMIT-REACHED (err u108))
+(define-constant ERR-MINT-PASS-LIMIT-REACHED (err u109))
+(define-constant ERR-ADD-MINT-PASS (err u110))
+
+
+(define-constant ERR-NOT-AUTHORIZED (err u401))
+(define-constant ERR-NOT-OWNER (err u402))
+(define-constant ERR-NOT-V1-OWNER (err u403))
+(define-constant ERR-NOT-ADMINISTRATOR (err u404))
```

Comment:
Does it make sense to use default labels for 4xx errors?
` (define-constant ERR-NOT-FOUND (err u404))`

Reply:
you mean ERR-NOT-ADMINISTRATOR should move to 5xx ?